hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/statemachine/commandhandler/ClosePipelineCommandHandler.java

```diff
@@ -68,12 +69,15 @@ public void handle(SCMCommand command, OzoneContainer ozoneContainer,
     final DatanodeDetails dn = context.getParent().getDatanodeDetails();
     ClosePipelineCommand closePipelineCommand = (ClosePipelineCommand) command;
     final PipelineID pipelineID = closePipelineCommand.getPipelineID();
+    final HddsProtos.PipelineID pipelineIdProto = pipelineID.getProtobuf();
 
     try {
       XceiverServerSpi server = ozoneContainer.getWriteChannel();
-      server.removeGroup(pipelineID.getProtobuf());
-      LOG.info("Close Pipeline {} command on datanode {}.", pipelineID,
-          dn.getUuidString());
+      if (server.isExist(pipelineIdProto)) {
+        server.removeGroup(pipelineIdProto);
+        LOG.info("Close Pipeline {} command on datanode {}.", pipelineID,
+            dn.getUuidString());
+      }
```

Comment:
I wonder if we should add an else branch for debugging (just in case we need it):

```
} else {
  LOG.debug("Ignoring close pipeline command for pipeline {} as it does not exist", pipelineID);
}
```

Reply:
@sodonnel this makes sense since this is a special case, will update, thanks~